mempool/metrics.go

```diff
@@ -3,32 +3,62 @@ package mempool
 import (
 	"github.com/go-kit/kit/metrics"
 	"github.com/go-kit/kit/metrics/discard"
-	prometheus "github.com/go-kit/kit/metrics/prometheus"
+	"github.com/go-kit/kit/metrics/prometheus"
 	stdprometheus "github.com/prometheus/client_golang/prometheus"
 )
 
+const MetricsSubsytem = "mempool"
+
 // Metrics contains metrics exposed by this package.
 // see MetricsProvider for descriptions.
 type Metrics struct {
 	// Size of the mempool.
 	Size metrics.Gauge
+	// Histogram of transaction sizes, in bytes.
+	TxSizeBytes metrics.Histogram
+	// Number of failed transactions.
+	FailedTxs metrics.Gauge
+	// Number of times transactions are rechecked in the mempool.
+	RecheckTimes metrics.Counter
 }
 
 // PrometheusMetrics returns Metrics build using Prometheus client library.
 func PrometheusMetrics(namespace string) *Metrics {
 	return &Metrics{
 		Size: prometheus.NewGaugeFrom(stdprometheus.GaugeOpts{
 			Namespace: namespace,
-			Subsystem: "mempool",
+			Subsystem: MetricsSubsytem,
 			Name:      "size",
 			Help:      "Size of the mempool (number of uncommitted transactions).",
 		}, []string{}),
+		TxSizeBytes: prometheus.NewHistogramFrom(stdprometheus.HistogramOpts{
+			Namespace: namespace,
+			Subsystem: MetricsSubsytem,
+			Name:      "tx_size_bytes",
+			Help:      "Transaction sizes in bytes.",
+			Buckets:   stdprometheus.ExponentialBuckets(1, 2, 10),
```

Comment:
what's the last bucket - 1024 bytes? some of our users have 100KB transactions. the max size for a transaction right now is 21MB (block size)